Error MSB4018 when building sqlproj with ref to sqlproj with large Post Deploy file.

### Environment
```
.NET SDK (reflecting any global.json):
 Version:   5.0.201
 Commit:    a09bd5c86c

Runtime Environment:
 OS Name:     Windows
 OS Version:  10.0.19042
 OS Platform: Windows
 RID:         win10-x64
 Base Path:   C:\Program Files\dotnet\sdk\5.0.201\

Host (useful for support):
  Version: 5.0.4
  Commit:  f27d337295

.NET SDKs installed:
  3.1.406 [C:\Program Files\dotnet\sdk]
  5.0.103 [C:\Program Files\dotnet\sdk]
  5.0.201 [C:\Program Files\dotnet\sdk]
```

### Instructions to reproduce.

1. Clone https://github.com/jovnas/sql
2. Run `dotnet build` in solution folder.

### Expected result
```
PS C:\Repos\Temp\sql> dotnet build
Microsoft (R) Build Engine version 16.9.0+57a23d249 for .NET
Copyright (C) Microsoft Corporation. All rights reserved.

  Determining projects to restore...
  All projects are up-to-date for restore.
  Using package name test1 and version 1.0.0
  Using SQL Server version Sql150
  Deleting existing file C:\Repos\Temp\sql\test1\obj\Debug\netstandard2.0\test1.dacpac
  Writing model to C:\Repos\Temp\sql\test1\obj\Debug\netstandard2.0\test1.dacpac
  Adding C:\Repos\Temp\sql\test1\Post-Deployment\Script.PostDeployment.sql to package
  test1 -> C:\Repos\Temp\sql\test1\bin\Debug\netstandard2.0\test1.dacpac
  test2 -> C:\Repos\Temp\sql\test2\bin\Debug\netstandard2.0\test2.dacpac

Build succeeded.
    0 Warning(s)
    0 Error(s)

Time Elapsed 00:00:03.08
```

### Actual result
```
PS C:\Repos\Temp\sql> dotnet build
Microsoft (R) Build Engine version 16.9.0+57a23d249 for .NET
Copyright (C) Microsoft Corporation. All rights reserved.

  Determining projects to restore...
  All projects are up-to-date for restore.
  Using package name test1 and version 1.0.0
  Using SQL Server version Sql150
  Deleting existing file C:\Repos\Temp\sql\test1\obj\Debug\netstandard2.0\test1.dacpac
  Writing model to C:\Repos\Temp\sql\test1\obj\Debug\netstandard2.0\test1.dacpac
  Adding C:\Repos\Temp\sql\test1\Post-Deployment\Script.PostDeployment.sql to package
  test1 -> C:\Repos\Temp\sql\test1\bin\Debug\netstandard2.0\test1.dacpac
C:\Program Files\dotnet\sdk\5.0.201\Microsoft.Common.CurrentVersion.targets(2182,5): error MSB4018: The "ResolveAssemblyReference" task failed unexpectedly. [C:\Repos\Temp\sql\test2\test2.csproj]
C:\Program Files\dotnet\sdk\5.0.201\Microsoft.Common.CurrentVersion.targets(2182,5): error MSB4018: System.InvalidOperationException: PE image does not have metadata. [C:\Repos\Temp\sql\test2\test2.csproj]
C:\Program Files\dotnet\sdk\5.0.201\Microsoft.Common.CurrentVersion.targets(2182,5): error MSB4018:    at System.Reflection.PortableExecutable.PEReader.GetMetadataBlock() [C:\Repos\Temp\sql\test2\test2.csproj]
C:\Program Files\dotnet\sdk\5.0.201\Microsoft.Common.CurrentVersion.targets(2182,5): error MSB4018:    at System.Reflection.Metadata.PEReaderExtensions.GetMetadataReader(PEReader peReader, MetadataReaderOptions options, MetadataStringDecoder utf8Decoder) [C:\Repos\Temp\sql\test2\test2.csproj]
C:\Program Files\dotnet\sdk\5.0.201\Microsoft.Common.CurrentVersion.targets(2182,5): error MSB4018:    at System.Reflection.Metadata.PEReaderExtensions.GetMetadataReader(PEReader peReader) [C:\Repos\Temp\sql\test2\test2.csproj]
C:\Program Files\dotnet\sdk\5.0.201\Microsoft.Common.CurrentVersion.targets(2182,5): error MSB4018:    at Microsoft.Build.Tasks.AssemblyInformation.CorePopulateMetadata() [C:\Repos\Temp\sql\test2\test2.csproj]
C:\Program Files\dotnet\sdk\5.0.201\Microsoft.Common.CurrentVersion.targets(2182,5): error MSB4018:    at Microsoft.Build.Tasks.AssemblyInformation.get_Dependencies() [C:\Repos\Temp\sql\test2\test2.csproj]
C:\Program Files\dotnet\sdk\5.0.201\Microsoft.Common.CurrentVersion.targets(2182,5): error MSB4018:    at Microsoft.Build.Tasks.AssemblyDependency.AssemblyMetadata..ctor(String path) [C:\Repos\Temp\sql\test2\test2.csproj]
C:\Program Files\dotnet\sdk\5.0.201\Microsoft.Common.CurrentVersion.targets(2182,5): error MSB4018:    at Microsoft.Build.Tasks.AssemblyInformation.<>c.<GetAssemblyMetadata>b__12_0(String p) [C:\Repos\Temp\sql\test2\test2.csproj]
C:\Program Files\dotnet\sdk\5.0.201\Microsoft.Common.CurrentVersion.targets(2182,5): error MSB4018:    at System.Collections.Concurrent.ConcurrentDictionary`2.GetOrAdd(TKey key, Func`2 valueFactory) [C:\Repos\Temp\sql\test2\test2.csproj]
C:\Program Files\dotnet\sdk\5.0.201\Microsoft.Common.CurrentVersion.targets(2182,5): error MSB4018:    at Microsoft.Build.Tasks.AssemblyInformation.GetAssemblyMetadata(String path, ConcurrentDictionary`2 assemblyMetadataCache, AssemblyNameExtension[]& dependencies, String[]& scatterFiles, FrameworkName& frameworkName) [C:\Repos\Temp\sql\test2\test2.csproj]
C:\Program Files\dotnet\sdk\5.0.201\Microsoft.Common.CurrentVersion.targets(2182,5): error MSB4018:    at Microsoft.Build.Tasks.SystemState.GetAssemblyMetadata(String path, ConcurrentDictionary`2 assemblyMetadataCache, AssemblyNameExtension[]& dependencies, String[]& scatterFiles, FrameworkName& frameworkName) [C:\Repos\Temp\sql\test2\test2.csproj]
C:\Program Files\dotnet\sdk\5.0.201\Microsoft.Common.CurrentVersion.targets(2182,5): error MSB4018:    at Microsoft.Build.Tasks.ReferenceTable.GetUnifiedAssemblyMetadata(Reference reference, IEnumerable`1& unifiedDependencies, String[]& scatterFiles) [C:\Repos\Temp\sql\test2\test2.csproj]
C:\Program Files\dotnet\sdk\5.0.201\Microsoft.Common.CurrentVersion.targets(2182,5): error MSB4018:    at Microsoft.Build.Tasks.ReferenceTable.FindDependenciesAndScatterFiles(Reference reference, List`1 newEntries) [C:\Repos\Temp\sql\test2\test2.csproj]
C:\Program Files\dotnet\sdk\5.0.201\Microsoft.Common.CurrentVersion.targets(2182,5): error MSB4018:    at Microsoft.Build.Tasks.ReferenceTable.FindAssociatedFiles() [C:\Repos\Temp\sql\test2\test2.csproj]
C:\Program Files\dotnet\sdk\5.0.201\Microsoft.Common.CurrentVersion.targets(2182,5): error MSB4018:    at Microsoft.Build.Tasks.ReferenceTable.ComputeClosure() [C:\Repos\Temp\sql\test2\test2.csproj]
C:\Program Files\dotnet\sdk\5.0.201\Microsoft.Common.CurrentVersion.targets(2182,5): error MSB4018:    at Microsoft.Build.Tasks.ReferenceTable.ComputeClosure(IEnumerable`1 remappedAssembliesValue, ITaskItem[] referenceAssemblyFiles, ITaskItem[] referenceAssemblyNames, List`1 exceptions) [C:\Repos\Temp\sql\test2\test2.csproj]
C:\Program Files\dotnet\sdk\5.0.201\Microsoft.Common.CurrentVersion.targets(2182,5): error MSB4018:    at Microsoft.Build.Tasks.ResolveAssemblyReference.Execute(FileExists fileExists, DirectoryExists directoryExists, GetDirectories getDirectories, GetAssemblyName getAssemblyName, GetAssemblyMetadata getAssemblyMetadata, GetLastWriteTime getLastWriteTime, GetAssemblyRuntimeVersion getRuntimeVersion, GetAssemblyPathInGac getAssemblyPathInGac, IsWinMDFile isWinMDFile, ReadMachineTypeFromPEHeader readMachineTypeFromPEHeader) [C:\Repos\Temp\sql\test2\test2.csproj]
C:\Program Files\dotnet\sdk\5.0.201\Microsoft.Common.CurrentVersion.targets(2182,5): error MSB4018:    at Microsoft.Build.Tasks.ResolveAssemblyReference.Execute() [C:\Repos\Temp\sql\test2\test2.csproj]
C:\Program Files\dotnet\sdk\5.0.201\Microsoft.Common.CurrentVersion.targets(2182,5): error MSB4018:    at Microsoft.Build.BackEnd.TaskExecutionHost.Microsoft.Build.BackEnd.ITaskExecutionHost.Execute() [C:\Repos\Temp\sql\test2\test2.csproj]
C:\Program Files\dotnet\sdk\5.0.201\Microsoft.Common.CurrentVersion.targets(2182,5): error MSB4018:    at Microsoft.Build.BackEnd.TaskBuilder.ExecuteInstantiatedTask(ITaskExecutionHost taskExecutionHost, TaskLoggingContext taskLoggingContext, TaskHost taskHost, ItemBucket bucket, TaskExecutionMode howToExecuteTask) [C:\Repos\Temp\sql\test2\test2.csproj]

Build FAILED.

C:\Program Files\dotnet\sdk\5.0.201\Microsoft.Common.CurrentVersion.targets(2182,5): error MSB4018: The "ResolveAssemblyReference" task failed unexpectedly. [C:\Repos\Temp\sql\test2\test2.csproj]
C:\Program Files\dotnet\sdk\5.0.201\Microsoft.Common.CurrentVersion.targets(2182,5): error MSB4018: System.InvalidOperationException: PE image does not have metadata. [C:\Repos\Temp\sql\test2\test2.csproj]
C:\Program Files\dotnet\sdk\5.0.201\Microsoft.Common.CurrentVersion.targets(2182,5): error MSB4018:    at System.Reflection.PortableExecutable.PEReader.GetMetadataBlock() [C:\Repos\Temp\sql\test2\test2.csproj]
C:\Program Files\dotnet\sdk\5.0.201\Microsoft.Common.CurrentVersion.targets(2182,5): error MSB4018:    at System.Reflection.Metadata.PEReaderExtensions.GetMetadataReader(PEReader peReader, MetadataReaderOptions options, MetadataStringDecoder utf8Decoder) [C:\Repos\Temp\sql\test2\test2.csproj]
C:\Program Files\dotnet\sdk\5.0.201\Microsoft.Common.CurrentVersion.targets(2182,5): error MSB4018:    at System.Reflection.Metadata.PEReaderExtensions.GetMetadataReader(PEReader peReader) [C:\Repos\Temp\sql\test2\test2.csproj]
C:\Program Files\dotnet\sdk\5.0.201\Microsoft.Common.CurrentVersion.targets(2182,5): error MSB4018:    at Microsoft.Build.Tasks.AssemblyInformation.CorePopulateMetadata() [C:\Repos\Temp\sql\test2\test2.csproj]
C:\Program Files\dotnet\sdk\5.0.201\Microsoft.Common.CurrentVersion.targets(2182,5): error MSB4018:    at Microsoft.Build.Tasks.AssemblyInformation.get_Dependencies() [C:\Repos\Temp\sql\test2\test2.csproj]
C:\Program Files\dotnet\sdk\5.0.201\Microsoft.Common.CurrentVersion.targets(2182,5): error MSB4018:    at Microsoft.Build.Tasks.AssemblyDependency.AssemblyMetadata..ctor(String path) [C:\Repos\Temp\sql\test2\test2.csproj]
C:\Program Files\dotnet\sdk\5.0.201\Microsoft.Common.CurrentVersion.targets(2182,5): error MSB4018:    at Microsoft.Build.Tasks.AssemblyInformation.<>c.<GetAssemblyMetadata>b__12_0(String p) [C:\Repos\Temp\sql\test2\test2.csproj]
C:\Program Files\dotnet\sdk\5.0.201\Microsoft.Common.CurrentVersion.targets(2182,5): error MSB4018:    at System.Collections.Concurrent.ConcurrentDictionary`2.GetOrAdd(TKey key, Func`2 valueFactory) [C:\Repos\Temp\sql\test2\test2.csproj]
C:\Program Files\dotnet\sdk\5.0.201\Microsoft.Common.CurrentVersion.targets(2182,5): error MSB4018:    at Microsoft.Build.Tasks.AssemblyInformation.GetAssemblyMetadata(String path, ConcurrentDictionary`2 assemblyMetadataCache, AssemblyNameExtension[]& dependencies, String[]& scatterFiles, FrameworkName& frameworkName) [C:\Repos\Temp\sql\test2\test2.csproj]
C:\Program Files\dotnet\sdk\5.0.201\Microsoft.Common.CurrentVersion.targets(2182,5): error MSB4018:    at Microsoft.Build.Tasks.SystemState.GetAssemblyMetadata(String path, ConcurrentDictionary`2 assemblyMetadataCache, AssemblyNameExtension[]& dependencies, String[]& scatterFiles, FrameworkName& frameworkName) [C:\Repos\Temp\sql\test2\test2.csproj]
C:\Program Files\dotnet\sdk\5.0.201\Microsoft.Common.CurrentVersion.targets(2182,5): error MSB4018:    at Microsoft.Build.Tasks.ReferenceTable.GetUnifiedAssemblyMetadata(Reference reference, IEnumerable`1& unifiedDependencies, String[]& scatterFiles) [C:\Repos\Temp\sql\test2\test2.csproj]
C:\Program Files\dotnet\sdk\5.0.201\Microsoft.Common.CurrentVersion.targets(2182,5): error MSB4018:    at Microsoft.Build.Tasks.ReferenceTable.FindDependenciesAndScatterFiles(Reference reference, List`1 newEntries) [C:\Repos\Temp\sql\test2\test2.csproj]
C:\Program Files\dotnet\sdk\5.0.201\Microsoft.Common.CurrentVersion.targets(2182,5): error MSB4018:    at Microsoft.Build.Tasks.ReferenceTable.FindAssociatedFiles() [C:\Repos\Temp\sql\test2\test2.csproj]
C:\Program Files\dotnet\sdk\5.0.201\Microsoft.Common.CurrentVersion.targets(2182,5): error MSB4018:    at Microsoft.Build.Tasks.ReferenceTable.ComputeClosure() [C:\Repos\Temp\sql\test2\test2.csproj]
C:\Program Files\dotnet\sdk\5.0.201\Microsoft.Common.CurrentVersion.targets(2182,5): error MSB4018:    at Microsoft.Build.Tasks.ReferenceTable.ComputeClosure(IEnumerable`1 remappedAssembliesValue, ITaskItem[] referenceAssemblyFiles, ITaskItem[] referenceAssemblyNames, List`1 exceptions) [C:\Repos\Temp\sql\test2\test2.csproj]
C:\Program Files\dotnet\sdk\5.0.201\Microsoft.Common.CurrentVersion.targets(2182,5): error MSB4018:    at Microsoft.Build.Tasks.ResolveAssemblyReference.Execute(FileExists fileExists, DirectoryExists directoryExists, GetDirectories getDirectories, GetAssemblyName getAssemblyName, GetAssemblyMetadata getAssemblyMetadata, GetLastWriteTime getLastWriteTime, GetAssemblyRuntimeVersion getRuntimeVersion, GetAssemblyPathInGac getAssemblyPathInGac, IsWinMDFile isWinMDFile, ReadMachineTypeFromPEHeader readMachineTypeFromPEHeader) [C:\Repos\Temp\sql\test2\test2.csproj]
C:\Program Files\dotnet\sdk\5.0.201\Microsoft.Common.CurrentVersion.targets(2182,5): error MSB4018:    at Microsoft.Build.Tasks.ResolveAssemblyReference.Execute() [C:\Repos\Temp\sql\test2\test2.csproj]
C:\Program Files\dotnet\sdk\5.0.201\Microsoft.Common.CurrentVersion.targets(2182,5): error MSB4018:    at Microsoft.Build.BackEnd.TaskExecutionHost.Microsoft.Build.BackEnd.ITaskExecutionHost.Execute() [C:\Repos\Temp\sql\test2\test2.csproj]
C:\Program Files\dotnet\sdk\5.0.201\Microsoft.Common.CurrentVersion.targets(2182,5): error MSB4018:    at Microsoft.Build.BackEnd.TaskBuilder.ExecuteInstantiatedTask(ITaskExecutionHost taskExecutionHost, TaskLoggingContext taskLoggingContext, TaskHost taskHost, ItemBucket bucket, TaskExecutionMode howToExecuteTask) [C:\Repos\Temp\sql\test2\test2.csproj]
    0 Warning(s)
    1 Error(s)

Time Elapsed 00:00:03.35
```

Please note that my files have Windows style new lines, so if git converts them to another new line style, it might work. In that case, just add a bunch of lines in `test1/Post-Deployment/DummyFile.sql`. It seems there is a file size limit at 39 256 bytes.

--- a/src/MSBuild.Sdk.SqlProj/Sdk/Sdk.targets
+++ b/src/MSBuild.Sdk.SqlProj/Sdk/Sdk.targets
@@ -136,6 +136,18 @@
   </Target>
 
   <!--
+    Removes any referenced .dacpac from the _ResolvedProjectReferencePaths items to ensure that the .dacpac
+    isn't passed to the ResolveAssemblyReferences target since that would treat it as a .NET assembly.
+  -->
+  <Target Name="RemoveDatabaseReferencesFromAssemblyReferences"
+          DependsOnTargets="ResolveDatabaseReferences"
+          AfterTargets="ResolveProjectReferences">
+    <ItemGroup>
+      <_ResolvedProjectReferencePaths Remove="%(DacpacReference.Identity)" />
+    </ItemGroup>
+  </Target>
+
+  <!--
     Gets a list of included files in the pre- and post-deploy scripts to include in the incremental build Inputs
   -->
   <Target Name="GetIncludedFiles" DependsOnTargets="ValidateEnvironment">

